src/steps/active-directory/converters.ts

```diff
@@ -0,0 +1,227 @@
+          intuneConfig?.mobileDeviceManagementAuthority,
+        intuneSubscriptionState: intuneConfig?.subscriptionState,
+      },
+    },
+  });
+}
+
+export function createMachineEntity(data: Machine): Entity {
+  return createIntegrationEntity({
+    entityData: {
+      source: {}, // removed due to size
+      assign: {
+        _class: entities.MACHINE._class,
+        _type: entities.MACHINE._type,
+        _key: data.id!,
+        id: data.id,
+        agentVersion: data.agentVersion,
+        defenderAvStatus: data.defenderAvStatus,
+        riskScore: data.riskScore,
+        name: data.computerDnsName,
+        computerDnsName: data.computerDnsName,
+        displayName: data.computerDnsName as string | undefined,
+        function: ['endpoint-protection', 'vulnerability-detection'],
+      },
+    },
+  });
+}
+
+export function createMachinesDeviceRelationship(data: Entity): Relationship {
+  return createMappedRelationship({
+    // source: data,
+    _class: RelationshipClass.MANAGES,
+    _type: MappedRelationships.MACHINE_MANAGES_DEVICE._type,
+    _mapping: {
+      sourceEntityKey: data._key,
+      relationshipDirection:
+        MappedRelationships.MACHINE_MANAGES_DEVICE.direction,
+      targetFilterKeys: [['_type', '_class', 'hostname']],
+      targetEntity: {
+        _class: TargetEntities.DEVICE._class,
+        _type: TargetEntities.DEVICE._type,
+        displayName: data.computerDnsName as string | undefined,
+        name: data.displayName,
+        id: data.id,
+        category: 'endpoint',
+        fqdn: data.computerDnsName as string | undefined,
+        hostname: data.computerDnsName as string | undefined,
+        osName: data.osPlatform as string | undefined,
+        osVersion: data.osVersion as string | undefined,
+        platform: data.osPlatform as string | undefined,
+      },
+      skipTargetCreation: true,
+    },
+  });
+}
+
+export function generateUserKey(user: User): string {
+  return user.id as string;
+}
+
+export function createUserEntity(data: UserLogon): Entity {
+  return createIntegrationEntity({
+    entityData: {
+      source: {}, // removed due to size
+      assign: {
+        _key: generateUserKey(data),
+        _class: entities.USER._class,
+        _type: entities.USER._type,
+        id: data.id,
+        name: data.accountName,
+        domain: data.accountDomain,
+        username: data.accountName,
+        displayName: data.accountName as string | undefined,
+        firstSeen: data.firstSeen,
+        lastSeen: data.lastSeen,
+        logonTypes: data.logonTypes,
+        active: data.isDomainAdmin,
+      },
+    },
+  });
+}
+
+export function createFindingEntity(defenderFinding: Finding): Entity {
+  return createIntegrationEntity({
+    entityData: {
+      source: {},
+      assign: {
+        id: defenderFinding.id,
+        _type: entities.FINDING._type,
+        _class: entities.FINDING._class,
+        _key: defenderFinding.id + '_' + Math.random() * 1000000,
```

Comment:
A `_key` value must be stable or the graph cannot be properly synchronized with the source IT system. Ideally the `_key` is a stable value that is part of the resource coming from the IT system. Is there anything on the finding that uniquely identifies the resource in Defender?